src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -750,8 +750,14 @@ internal static bool MatchesModulePath(string modulePath, string requiredPath)
                 return moduleName;
             }
 
+#if UNIX
+    // On *nix, Path.AltDirectorySeparatorChar is '/', but PowerShell also supports '\\' as a dir separator
+    const char altDirSeparatorChar = '\\';
+#else
+    const char altDirSeparatorChar = Path.AltDirectorySeparatorChar;
+#endif
             // Standardize directory separators
-            moduleName = moduleName.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+            moduleName = moduleName.Replace(altDirSeparatorChar, Path.DirectorySeparatorChar);
```

Comment:
I expect `GetResolvedPath` will take care of the path normalization, but need to verify that. If that's the case, we don't need this path separator replacement.
And, in case we really need this, please use the existing `StringLiterals.DefaultPathSeparator` and `StringLiterals.AlternatePathSeparator`.